code/modules/mob/living/carbon/status_procs.dm

```diff
@@ -63,6 +63,43 @@
 		clear_fullscreen("high")
 		clear_alert("high")
 
+/mob/living/carbon/adjust_disgust(amount)
+	var/old_disgust = disgust
+	if(amount>0)
+		disgust = min(disgust+amount, DISGUST_LEVEL_MAXEDOUT)
+		switch(disgust)
```

Comment:
NO! Make updating the HUD a proc that happens on life! Even better, attach disgust to the stomach and make it update on the stomach's on_life proc. Don't have this copypasta stuff on adjust_disgust and set_disgust. I say add it to the stomach and then if a species or someone doesn't have a stomach, they can't be disgusted.